### ISSUE TITLE
Add checks for OpenJDK Style §4.1 - Package Names

### ISSUE THREAD

**Issue by MiriyalaJayanth-19** [checkstyle/checkstyle#19640]
Parent issue: #19604

<img width="988" height="119" alt="Image" src="https://github.com/user-attachments/assets/99e4fdab-c93a-43c9-b937-5fab7bb7bdf9" />

## Description
Add Checkstyle checks for all rules under §4.1 - Package Names of the OpenJDK Java Style Guidelines v6.

## Rules covered
[From §4.1 - Package Names](https://cr.openjdk.org/~alundblad/styleguide/index-v6.html#toc-package-names):

1) Package names should be all lower case without underscores or other special characters.
2) Don’t use plural form. Follow the convention of the standard API which uses for instance `java.lang.annotation` and not `java.lang.annotations`. [Not coverable]

## Checks to add to openjdk_checks.xml
TreeWalker-level:
```xml
<module name="PackageName">
  <property name="format" value="^[a-z]+(\.[a-z][a-z0-9]*)*$"/>
</module>

**Comment by vivek-0509** [comment]
@MiriyalaJayanth-19 the issue is for open jdk style guide you have given the reference of google style, please mention exact rules mentioned in the guide

**Comment by MiriyalaJayanth-19** [comment]
My bad, thanks for clarification . Did the changes

**Comment by vivek-0509** [comment]
we need to use this configuration to enforce those rules not the default config
```
<module name="PackageName">
  <property name="format" value="^[a-z]+(\.[a-z][a-z0-9]*)*$"/>
</module>

```

**Comment by MiriyalaJayanth-19** [comment]
Yeah I meant that . Now i edited in the description. Thank you for pointing it .

**Comment by vivek-0509** [comment]
@MiriyalaJayanth-19 Now it looks good, do follow chapterwise format not module wise take reference from google style, wait for this [PR](https://github.com/checkstyle/checkstyle/pull/19642) to get merged than you can start working

**Comment by vivek-0509** [comment]
@MiriyalaJayanth-19 the pr is merged you can make your PR

**Comment by MiriyalaJayanth-19** [comment]
@vivek-0509 the pr is open could you please review it

**Comment by vivek-0509** [comment]
@MiriyalaJayanth-19  please reference the issue in the PR

**Comment by MiriyalaJayanth-19** [comment]
@vivek-0509 i had refered the pr

**Comment by romani** [comment]
> Don’t use plural form.

I do not think we can cover it, as we cannot recognized words, we need dictionary for this, we do not have it, so lets mark this part not coverable.

**Comment by MiriyalaJayanth-19** [comment]
Done.

**Comment by romani** [comment]
Fix is merged
